src/SSHDebugPS/PSOutputParser.cs

```diff
@@ -147,7 +147,7 @@ private bool ProcessHeaderLine(/*OPTIONAL*/ string headerLine)
             SkipNonWhitespace(headerLine, ref index);
 
             // make sure the line is now empty, aside from whitespace
-            if (SkipWhitespace(headerLine, ref index))
+            if (SkipWhitespace(headerLine, ref index) && index < strLen)
```

Comment:
Why did you add the condition? `SkipWhitespace` will return false if there is no more whitespace before the end.